public/app/plugins/panel/timeseries/utils.ts

```diff
@@ -137,7 +137,7 @@ export function prepareGraphableFields(
 
     const frameFields = nullToValue(nulledFrame).fields;
 
-    for (let fieldIdx = 0; fieldIdx < frameFields?.length ?? 0; fieldIdx++) {
+    for (let fieldIdx = 0; fieldIdx < frameFields?.length; fieldIdx++) {
```

Comment:
this one seems wrong. maybe it wasnt doing what i thought it was doing due to operator precedence.

it should probably be `fieldIdx < (frameFields?.length ?? 0);`